CONTRIBUTING.md

```diff
@@ -1,28 +1,47 @@
+When you write unit tests, please follow our naming conventions for unit tests. Please include a top level `describe('ClassName')` block with the name of the class you're testing. Inside that `describe()` block, create another `describe('#methodOne()')` block for each method you have in your class.
 
-2. To ACTUALLY run the tests, do this at the root of the project:  
-npm test
+For each method include a separate `it('should do something')` test case for each logical edge case in your new changes. After you write couple of tests, run the tests, check the code coverage and make sure all lines of code are covered, if not, just add more test cases that covers it. For reference and inspiration, please check our `tests` directory.
+
+Make sure the description of the blocks matches the format you see here.
+
+### Code Style Guidelines
+Internally, we're using ESlint to check for codestyle issues using the airbnb preset. To stay consistent, please make sure you follow this codestyle carefully and that your changes doesn't have any styling issues. We can't accept your PR if your changes are not consistent with our code style.
+
+You can check for styling issues in a file using ESlint using the following command: (in the root of our repo)
+
+```
+./node_modules/eslint/bin/eslint.js filename.js
+```
+You'll probably find ESlint plugins for your favorite code editor that will make this easier.
+
+## Opening Issues
+Before you open an issue, please make sure it's not a duplicate of an existing open (or closed) issue. A Github issue could be a feature request, a bug report, or simply a discussion about related topics. For support questions, please use StackOverflow with the `serverless` tag instead.
+
+When you open an issue, we'll add relevant labels to that issue asap to reflect category, area of code, status and priority of the issue. We'll keep those labels updated to keep you updated on our progress. 
+
+If your issue is a bug report, please make sure you include an accurate/detailed description of the issue you're facing and how to reproduce it. Make sure you mention your Serverless version, as well as your environment (node/npm version, OS...etc)
+
+## Providing Support
+This simplest thing you can do to help us move forward and make an impact on our progress is to simply provide support to other people having some difficulties with their Serverless project. You can do that by replying to issues on Github, or simply chatting with other community members on Gitter. Your very presence is enough to make us happy.
+
+## Improving Documentation
+Maintaining and updating the docs on a regular basis is a hard task. The more eyeballs on the docs the higher quality it'll get and the less chances there will be for typos and confusion. We keep our docs in our `serverless-docs` repo. If you see any issues with our docs, simply open an issue or a PR.
```

Comment:
We keep documentation in the docs folder, not a separate repo. After that change GTM